internal/api/handlers/utils.go

```diff
@@ -138,3 +138,57 @@ func FetchCollectible(
 		TokenId:  tokenId,
 	}, nil
 }
+
+/*
+NOTE: This function is not part of SEP-50 and should only be used with Meridian Pay NFTs or contracts that extend the standard.
+*/
+func FetchOwnerTokens(
+	rpc types.RPCService,
+	ctx context.Context,
+	accountId *txnbuild.SimpleAccount,
+	contractID string,
+	owner string,
+) ([]string, error) {
+
+	id, err := utils.ScAddressFromString(contractID)
```

Comment:
just a thought: is it possible to move common functionality between this function and `fetchCollectible` to a separate function? This way the individual functions would become easier to read.